google/cloud/spanner/doc/spanner-main.dox

```diff
@@ -24,164 +24,65 @@ using the C++ client library.
-```
+## API Notes
 
-#### Integrating with Bazel
+The following are general notes about using the API.
 
-As we do not currently have releases of `google-cloud-cpp-spanner` you need to
-select the SHA-1 of the
-[commit](https://github.com/googleapis/google-cloud-cpp-spanner/commits/master)
-you want to use.
+### Environment Variables
 
-In your [WORKSPACE][workspace-definition-link] file add a dependency to download
-and install the library, for example:
-
-@code {.py}
-# Add the necessary Starlark functions to fetch google-cloud-cpp-spanner.
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-# Change the version and SHA256 hash as needed.
-http_archive(
-    name = "com_github_googleapis_google_cloud_cpp_spanner",
-    sha256 = "a833d3c1a6d127132e961350829babac521b62b4c837b88d7c219b400e98fed1",
-    strip_prefix = "google-cloud-cpp-spanner-0.8.0",
-    url = "https://github.com/googleapis/google-cloud-cpp-spanner/archive/v0.8.0.tar.gz",
-)
-@endcode
+There are several environment variables that can be set to configure certain
+behaviors in the library.
 
-Then load the dependencies of the library:
+- `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` turns on logging in the library, basically
+  the library always "logs" but the logging infrastructure has no backend to
+  actually print anything until the application sets a backend or they set this
+  environment variable.
 
-```{.py}
-load("@com_github_googleapis_google_cloud_cpp_spanner//bazel:google_cloud_cpp_spanner_deps.bzl", "google_cloud_cpp_spanner_deps")
+- `GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc` turns on tracing for most gRPC
+  calls, the library injects an additional Stub decorator that prints each gRPC
+  request and response.
 
-google_cloud_cpp_spanner_deps()
+- `GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc-streams` turns on tracing for streaming
+  gRPC calls, such as `Client::Read()`, `Client::ExecutEQuery()`, and
+  `Client::ProfileQuery()`. This can produce a lot of output, so use with
+  caution!
 
-# Configure @com_google_googleapis to only compile C++ and gRPC:
-load("@com_google_googleapis//:repository_rules.bzl", "switched_rules_by_language")
+- `GOOGLE_CLOUD_PROJECT=...` is used in examples and integration tests to
+  configure the GCP project.
 
-switched_rules_by_language(
-    name = "com_google_googleapis_imports",
-    cc = True,
-    grpc = True,
-)
+- `GOOGLE_CLOUD_CPP_SPANNER_INSTANCE=...` is used in examples and integration
+  tests to, well, configure the spanner instance.
 
-# You have to manually call the corresponding function for google-cloud-cpp and
-# gRPC:  https://github.com/bazelbuild/bazel/issues/1550
-load("@com_github_googleapis_google_cloud_cpp_common//bazel:google_cloud_cpp_deps.bzl", "google_cloud_cpp_deps")
+- `GOOGLE_CLOUD_CPP_SPANNER_IAM_TEST_SA=...` is used in examples and
+  integration tests to set the service account for testing IAM operations.
 
-google_cloud_cpp_deps()
+- `SPANNER_EMULATOR_HOST=host:port` tells the library to connect to the
```

Comment:
And GOOGLE_CLOUD_CPP_SPANNER_DEFAULT_ENDPOINT, which is subtly diffferent.

Reply:
Done. Added.

FTR, I originally omitted that one on purpose because it seems like something that is only useful for us when testing w/ internal versions of Spanner. I don't think users would ever have a need to set it. But documenting it here is fine with me.